### PR TITLE
feat: implement XdgIdentityService

### PR DESCRIPTION
**Pull request by d-loose**

### CODE REVIEW

File: packages/ubuntu_desktop_installer/test/services/identity_service_test.dart [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+import 'identity_service_test.mocks.dart';
+
+@GenerateMocks([DBusRemoteObject, DBusClient])
+void main() {
+  const testIdentity = IdentityData(
+    realname: 'Arthur Dent',
+    username: 'adent',
+    hostname: 'heart-of-gold',
+  );
+  group('subiquity', () {
```
Comment (jpnurmi) [inline packages/ubuntu_desktop_installer/test/services/identity_service_test.dart:17]: thanks! :)

**Review by jpnurmi — APPROVED** [review]
LGTM :+1:

File: packages/ubuntu_desktop_installer/lib/services/identity_service.dart [hunk 2]
```diff
@@ -27,3 +30,107 @@ class SubiquityIdentityService implements IdentityService {
...
+          'Modifying the default user is not yet implemented');
+    }
+
+    await _accountObject.callMethod(
+      'org.freedesktop.Accounts',
+      'CreateUser',
+      [
+        DBusString(_identity!.username),
+        DBusString(_identity!.realname),
+        const DBusUint32(1), // Administrator account
```
Comment (jpnurmi) [inline packages/ubuntu_desktop_installer/lib/services/identity_service.dart:120]: should it be `DBusInt32`?

image

image

Comment (d-loose) [reply packages/ubuntu_desktop_installer/lib/services/identity_service.dart:120]: Oh, yes it should be - sorry!

**Pull request merged**